File: src (cpp)/infrastructure/facility/park.cpp
#include "park.h"

Park::Park(bool accessible, const std::string& _name): Facility(accessible), def_facility_type("Park"), name(_name) {
  facility_type = def_facility_type;
}
Park::Park(const Park& P): Facility(P.is_accessible), def_facility_type("Park"), name(P.name) {
  facility_type = P.facility_type;
  is_accessible = P.is_accessible;
}
Park& Park::operator=(const Park& P) {
  name = P.name;
  facility_type = P.facility_type;
  is_accessible = P.is_accessible;
  return *this;
}
std::string Park::GetParkName() {
  return name;
}
char Park::Render() {
  return '*';
}

File: src/virtualzoo/infrastructure/facility/Park.java
package virtualzoo.infrastructure.facility;

/**
 * Kelas Park mendefinisikan sebuah taman di dalam Zoo.
 *
 * @author Oktavianus Handika - 13515035
 * @version 3.0
 * @since 3.0
 */
public class Park extends Facility {

  /**
   * Nama untuk setiap Park.
   */
  protected String name;

  /**
   * Constructor
   *
   * Membuat sebuah taman.
   */
  public Park(boolean accessible, String facilityName) {
    super(accessible);
    facilityType = "Park";
    name = facilityName;
  }

  /**
   * Mengembalikan suatu char yang menandakan Park dalam satu Cell.
   */
  public char render() {
    return '*';
  }
}
